app/src/main/java/io/appium/uiautomator2/handler/GetClipboard.java

```diff
@@ -71,4 +69,30 @@ public AppiumResponse safeHandle(IHttpRequest request) {
             return new AppiumResponse(getSessionId(request), WDStatus.UNKNOWN_ERROR, e);
         }
     }
+
+    // Clip feature should run with main thread
+    private AppiumResponse getClipboardTextResponse(final IHttpRequest request) {
+        AppiumGetClipRunnerble runnable = new AppiumGetClipRunnerble(request);
+        mInstrumentation.runOnMainSync(runnable);
+        return runnable.getResponse();
+    }
+
+    private class AppiumGetClipRunnerble implements Runnable {
+        private IHttpRequest request;
+        private AppiumResponse response;
+
+        AppiumGetClipRunnerble(IHttpRequest request) {
+            this.request = request;
+        }
+
+        @Override
+        public void run() {
+            response = new AppiumResponse(getSessionId(request), WDStatus.SUCCESS,
```

Comment:
@KazuCocoa can we here return base64-encoded content and wrap it into AppiumResponse inside handle method? Just for the sake of common code style.

Reply:
Also ```switch(contentType)``` can be moved to this ```Runnable```.